internal/cli/plugin/plugin_asset.go

```diff
@@ -226,7 +225,7 @@ func extractTarGz(src string, dest string) error {
 					return errCopyFileContentExtraction
 				}
 			}
-
+			outFile.Close()
```

Comment:
Wouldn't it be better to put this in a `defer` statement?
You will not close the file when you throw `errCopyFileContentExtraction`

Reply:
The linter threw me an error saying that I should not use defer in a loop

Reply:
Can you close the file before returning the error too?